### ISSUE TITLE
Confusion about the Loss function L_{atk}

### ISSUE THREAD

**Issue by harshakokel** [danielzuegner/gnn-meta-attack#6]
Hello,

I am confused about the loss function in the paper Zuegner et al ICLR 2019. 

In the paper, you mention that <img width="100" alt="portfolio_view" src="https://user-images.githubusercontent.com/858059/80314899-007bc680-87ba-11ea-8f24-d14a4e8bddb7.png"> or <img width="100" alt="portfolio_view" src="https://user-images.githubusercontent.com/858059/80314919-21dcb280-87ba-11ea-8abb-869009b96b3e.png"> and then gradient is 
<img width="200" alt="portfolio_view" src="https://user-images.githubusercontent.com/858059/80315036-cd860280-87ba-11ea-803e-439b115ce4ff.png">

But, in the algorithm the gradient equation is missing the -ve sign.

![image](https://user-images.githubusercontent.com/858059/80314864-c27ea280-87b9-11ea-9c65-6cb2b3702cb8.png)

I am also confused about the same thing in the code. For 'Meta-Train' variant, I think the gradient calculation [here](https://github.com/danielzuegner/gnn-meta-attack/blob/ab0bd077d790cb63943d75c15e538ac8fdcb8698/metattack/meta_gradient_attack.py#L363) is trying to minimize the classification error of the examples in the training [here](https://github.com/danielzuegner/gnn-meta-attack/blob/ab0bd077d790cb63943d75c15e538ac8fdcb8698/metattack/meta_gradient_attack.py#L348), instead of trying to maximize the classification error. 

What am I missing?

**Comment by danielzuegner** [comment]
Hi, thanks for your message. 

Sorry about the confusion. Since the attacker is trying to maximize the classification loss, Algorithm 1 is correct and in the paper it should read `L_atk=L_self`. Let me know if that helps!

Best,

Daniel

**Comment by harshakokel** [comment]
Thank you. This helps.
